### PR TITLE
[codex] optimize CUDA prefill and add SpecPrefill benchmarks

### PR DESCRIPTION
**Pull request by DeanoC**
## Summary

- Optimize CUDA prefill paths with cuBLAS-backed BF16 prefill matmuls, reduced unconditional synchronizations, and lower memset overhead for Qwen3.5/Gemma prefill.
- Enable CUDA Qwen3.6 cross-family SpecPrefill with Qwen3.5-0.8B cosine scoring, release-bake-backed INT4 usage, and a conservative default keep ratio of `0.75` on sm86.
- Add the `supersonic-bench` crate, benchmark matrix/parity checks, sm86 profiling scripts, and a Python oracle quality harness for SpecPrefill logits.
- Document the quality/performance lanes: `Spec075` is the default quality-preserving lane, while `Spec050` remains an explicit balanced/perf lane.

## Measured Results

- Qwen3.5-0.8B CUDA prefill improved from roughly `1040 ms` to roughly `672 ms` on the 533-token test prompt.
- Qwen3.6-35B-A3B INT4 standard sm86 gate:
  - dense INT4: `4635.9 ms`
  - `int4-spec075`: `3591.0 ms`
  - quality over 7 prompts: `argmax_match_rate=1.000`, `cosine_min=0.906974`, `top5_min=3`, threshold pass.

## Validation

- `SUPERSONIC_BACKENDS=cuda cargo test -p runner --bin supersonic` (`71 passed`)
- `cargo test -p supersonic-bench`
- `python3 -m pytest oracle/bench/tests` (`28 passed`)
- `SUPERSONIC_BACKENDS=cuda cargo test -p runner --test bench_combo_parity`
- `PROMPT_SET=standard MEASUREMENT_RUNS=1 RUN_ROOT=target/bench-runs-sm86-conservative-standard tests/sm86/bench_specprefill_quality.sh /workspace/models/Qwen3.6-35B-A3B-FP8 /workspace/models/Qwen3.5-0.8B`
- `git diff --check`

## Notes

- The attempted Qwen3.6 dense token-loop prefill path remains gated behind `SUPERSONIC_QWEN36_DENSE_PREFILL_TOKEN_LOOP=1` because it was correctness-checked but slower than the default path.
- The previous `CACHE_POS_INHERIT` warning was a redundant runner-side re-export, not a behavior bug; the actual FFI sentinel remains in use.

### CODE REVIEW

**Comment by DeanoC** [comment]
@codex review

**Comment by DeanoC** [comment]
@codex review
